apps/webapp/src/state/helpers.ts

```diff
@@ -1,18 +1,54 @@
 import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
 import { viewClient } from '../clients/grpc.ts';
 import { uint8ArrayToHex } from '@penumbra-zone/types';
+import { TransactionId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/txhash/v1alpha1/txhash_pb';
+import { sha256Hash } from '@penumbra-zone/crypto-web';
 
-export const planWitnessBuildBroadcast = async (req: TransactionPlannerRequest) => {
-  const { plan } = await viewClient.transactionPlanner(req);
-  if (!plan) throw new Error('no plan in response');
+export const planWitnessBuildBroadcast = async (
+  req: TransactionPlannerRequest,
+  awaitDetection = true,
+) => {
+  const { plan: transactionPlan } = await viewClient.transactionPlanner(req);
 
-  const { transaction } = await viewClient.authorizeAndBuild({
-    transactionPlan: plan,
-  });
-  if (!transaction) throw new Error('no transaction in response');
+  if (!transactionPlan) throw new Error('no plan in response');
 
-  const { id } = await viewClient.broadcastTransaction({ transaction, awaitDetection: true });
-  if (!id) throw new Error('no id in broadcast response');
+  let transaction;
```

Comment:
```suggestion
  let transaction: Transaction | undefined;
```

Reply:
linter and type system are already happy with it

Reply:
This feels like a linter issue 🤔 . `let transaction;` would qualify as an `any` no? We should investigate why this was allowed through.

Reply:
Ah, this actually _is_ allowed with our current config. Going to open a new PR adding a lint rule to catch these.